services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -318,6 +320,10 @@ public BeaconChainController(
     this.receivedBlockEventsChannelPublisher =
         eventChannels.getPublisher(ReceivedBlockEventsChannel.class);
     this.forkChoiceExecutor = new AsyncRunnerEventThread("forkchoice", asyncRunnerFactory);
+    this.debugDataDumper =
+        new DebugDataDumper(
+            serviceConfig.getDataDirLayout().getDebugDataDirectory(),
+            beaconConfig.p2pConfig().isP2pDumpsToFileEnabled());
```

Comment:
Potentially the more 'normal' implementation would be to use the Noop implementation here, rather than have this disabled/enabled flag going through the structure. 
It'd improve the overall flow, and the small difference is that if P2pDumps is enabled, use the real one, otherwise the noop.